File: ambeth/Ambeth.IoC/ambeth/converter/StringToLongArrayConverter.cs
using De.Osthus.Ambeth.Log;
using De.Osthus.Ambeth.Util;
using System;
using System.Text;

namespace De.Osthus.Ambeth.Converter
{
    public class StringToLongArrayConverter : IDedicatedConverter
    {
        [LogInstance]
        public ILogger Log { private get; set; }

        public Object ConvertValueToType(Type expectedType, Type sourceType, Object value, Object additionalInformation)
        {
            if (typeof(long[]).Equals(expectedType))
            {
                String[] split = StringToPatternConverter.splitPattern.Split((String)value);
                long[] result = new long[split.Length];
                for (int a = split.Length; a-- > 0; )
                {
                    result[a] = Int64.Parse(split[a]);
                }
                return result;
            }
            StringBuilder sb = new StringBuilder();
            long[] array = (long[])value;
            foreach (long item in array)
            {
                if (sb.Length > 0)
                {
                    sb.Append(StringToPatternConverter.splitPattern.ToString());
                }
                sb.Append(item);
            }
            return sb.ToString();
        }
    }
}

File: jambeth/jambeth-ioc/src/main/java/de/osthus/ambeth/converter/StringToLongArrayConverter.java
package de.osthus.ambeth.converter;

import de.osthus.ambeth.util.IDedicatedConverter;

public class StringToLongArrayConverter implements IDedicatedConverter
{
	@Override
	public Object convertValueToType(Class<?> expectedType, Class<?> sourceType, Object value, Object additionalInformation)
	{
		if (long[].class.equals(expectedType))
		{
			String[] split = StringToPatternConverter.splitPattern.split((String) value);
			long[] result = new long[split.length];
			for (int a = split.length; a-- > 0;)
			{
				result[a] = Long.parseLong(split[a]);
			}
			return result;
		}
		StringBuilder sb = new StringBuilder();
		long[] array = (long[]) value;
		for (long item : array)
		{
			if (sb.length() > 0)
			{
				sb.append(StringToPatternConverter.splitPattern.pattern());
			}
			sb.append(item);
		}
		return sb.toString();
	}
}
